webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/OrderedScrambles.java

```diff
@@ -0,0 +1,245 @@
+                        Collections.sort(scrambleRequestListByDay.get(index));
+                        
+                        String pdfFileName = "Printing/Ordered Scrambles/";
+
+                        if (hasMultipleVenues) {
+                            pdfFileName += venueName.replace('/', ' ')+"/";
+                        }
+
+                        if (hasMultipleDays || dayList.size() > 1) {
+                            pdfFileName += "Day "+dayList.get(index)+"/";
+                        }
+
+                        pdfFileName += "Ordered Scrambles";
+                        
+                        // In addition to different folders, we stamp venue, day and room in the PDF's name
+                        // to prevent different files with the same name.
+                        if (hasMultipleVenues) {
+                            pdfFileName += " - " + venueName.replace('/', ' ');
+                        }
+                        if (hasMultipleDays || dayList.size() > 1) {
+                            pdfFileName += " - Day "+dayList.get(index);
+                        }
+                        
+                        if (hasMultipleRooms) {
+                            pdfFileName += " - "+roomName.replace('/', ' ');
+                        }
+                        pdfFileName += ".pdf";
+                        
+                        // removing slashes may add double+ extra spaces
+                        pdfFileName = pdfFileName.replaceAll("\\s+", " ");
+
+                        parameters.setFileNameInZip(pdfFileName);
+                        zipOut.putNextEntry(null, parameters);
+                        ScrambleRequest[] scrambleRequests = scrambleRequestListByDay.get(index).toArray(new ScrambleRequest[scrambleRequestListByDay.get(index).size()]);
+                        ByteArrayOutputStream baos = ScrambleRequest.requestsToPdf(globalTitle, generationDate, scrambleRequests, null);
+                        zipOut.write(baos.toByteArray());
+                        zipOut.closeEntry();
+                    }
+                }
+            }
+        }
+    }
+
+    private static boolean compareFirstCharToNumber(String letter, int number) {
+        return letter.charAt(0)-'A' == number-1;
+    }
+
+    private static String removeQuotation(String s) {
+        if (s.charAt(0) == '"') {
+            s = s.substring(1, s.length());
+        }
+        if (s.charAt(s.length()-1) == '"') {
+            s = s.substring(0, s.length()-1);
+        }
+        return s;
+    }
+
+    // In case venue or room is using markdown
+    private static String parseMarkdown(String s) {
+        if (s.indexOf('[') >= 0 && s.indexOf(']') >= 0) {
+            s = s.substring(s.indexOf('[')+1, s.indexOf(']'));
+        }
+        return s;
+    }
+
+    private static String getEarlierActivityString(JsonObject scheduleJson) {
+
+        DateTime date = null;
+        String out = "";
```

Comment:
nit: I think it's cleaner to set this to `null` initially (don't forget to update the check down on line 242 as well)

Reply:
Fixed